src/main/java/io/anserini/index/IndexReaderUtils.java

```diff
@@ -375,6 +375,36 @@ public static Document document(IndexReader reader, String docid) {
     }
   }
 
+  /**
+   * Fetches the Lucene {@link Document} based on some field other than its unique collection docid.
+   * For example, scientific articles might have DOIs.
+   * The method is named to be consistent with Lucene's {@link IndexReader#document(int)}, contra Java's standard
+   * method naming conventions.
+   *
+   * @param reader index reader
+   * @param field field
+   * @param id unique id
+   * @return corresponding Lucene {@link Document} based on the value of a specific field
+   */
+  public static Document documentByField(IndexReader reader, String field, String id) {
+    try {
+      IndexSearcher searcher = new IndexSearcher(reader);
+      Query q = new TermQuery(new Term(field, id));
+      TopDocs rs = searcher.search(q, 1);
+      ScoreDoc[] hits = rs.scoreDocs;
+
+      if (hits == null || hits.length == 0) {
+        // Silently eat the error and return -1
```

Comment:
maybe remove/change this comment as this is not exactly an error?

Reply:
Yup!